src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -128,11 +128,21 @@ private async Task StartDashboardAsync(CancellationToken cancellationToken = def
         var dashboardExecutableSpec = new ExecutableSpec
         {
             ExecutionType = ExecutionType.Process,
-            ExecutablePath = "dotnet",
-            Args = [fullyQualifiedDashboardPath],
             WorkingDirectory = dashboardWorkingDirectory
         };
 
+        if (StringComparer.OrdinalIgnoreCase.Equals(".dll", Path.GetExtension(fullyQualifiedDashboardPath)))
+        {
+            // The dashboard path is a DLL, so run it with `dotnet <dll>`
+            dashboardExecutableSpec.ExecutablePath = "dotnet";
+            dashboardExecutableSpec.Args = [fullyQualifiedDashboardPath];
+        }
+        else
+        {
+            // Assume the dashboard path is directly executable
+            dashboardExecutableSpec.ExecutablePath = fullyQualifiedDashboardPath;
+        }
```

Comment:
The idea here is that we cannot use `.exe` to check for executables in a cross-platform way, so we check for `.dll` and assume anything else is executable.